csharp/test/IceSSL/configuration/AllTests.cs

```diff
@@ -878,6 +762,83 @@ public static Test.ServerFactoryPrx allTests(Test.TestHelper helper, string test
             }
             Console.Out.WriteLine("ok");
 
+            Console.Out.Write("testing hot certificate reloading... ");
+            Console.Out.Flush();
+            {
+                using var serverCertificate1 = new X509Certificate2($"{defaultDir}/s_rsa_ca1.p12", "password");
+                using var serverCertificate2 = new X509Certificate2($"{defaultDir}/s_rsa_ca2.p12", "password");
+                int count = 0;
+                using Ice.Communicator serverCommunicator = Ice.Util.initialize();
+                Ice.ObjectAdapter adapter = serverCommunicator.createObjectAdapterWithEndpoints(
+                    "Server",
+                    $"ssl -h {defaultHost}",
+                    new SslServerAuthenticationOptions
+                    {
+                        ServerCertificateSelectionCallback = (object sender, string hostName) =>
+                        {
+                            return count++ == 0 ? serverCertificate1 : serverCertificate2;
```

Comment:
The server starts using "s_rsa_ca1" certificate and switch to "s_rsa_ca2" after the initial connection.